include/z64camera.h

```diff
@@ -0,0 +1,389 @@
+#ifndef Z64CAMERA_H
+#define Z64CAMERA_H
+
+#include "ultra64.h"
+
+// these two angle conversion macros are slightly inaccurate
+#define CAM_DEG_TO_BINANG(degrees) (s16)((degrees) * 182.04167f + .5f)
+#define CAM_BINANG_TO_DEG(binang) ((f32)(binang) * (360.0001525f / 0xFFFF))
+
+#define NUM_CAMS 4
+
+// Camera IDs are indices into `cameraPtrs`
+#define CAM_ID_MAIN 0 // The index of the main camera
+#define CAM_ID_SUB_FIRST 1 // The index sub cameras start at
+#define CAM_ID_NONE -1 // Used to indicate no camera. Can be used to default to the active camera in some scenarios
+
+#define SUB_CAM_ID_DONE 0 // Used in some actors for variables holding sub camera IDs to indicate "subcam is finished"
+
+#define CAM_TRACKED_PLAYER(camera) ((Player*)camera->trackActor)
+
+// Camera behaviorFlags. Flags specifically for settings, modes, and bgCam
+// Used to store current state, only CAM_BEHAVIOR_SETTING_1 and CAM_BEHAVIOR_BG_2 are read from and used in logic
+// Setting (0x1, 0x10)
+#define CAM_BEHAVIOR_SETTING_USE_PRIORITY (1 << 0) // Use settings priority system
+#define CAM_BEHAVIOR_SETTING_2 (1 << 4)
+// Mode (0x2, 0x20)
+#define CAM_BEHAVIOR_MODE_1 (1 << 1)
+#define CAM_BEHAVIOR_MODE_2 (1 << 5)
+// bgCam (0x4, 0x40)
+#define CAM_BEHAVIOR_BGCAM_1 (1 << 2)
+#define CAM_BEHAVIOR_BGCAM_2 (1 << 6)
+
+// Camera stateFlags. Variety of generic flags
+#define CAM_STATE_0 (1 << 0) // Must be set for the camera from changing settings based on the bg surface
+#define CAM_STATE_1 (1 << 1)
+#define CAM_STATE_2 (1 << 2)
+#define CAM_STATE_3 (1 << 3)
+#define CAM_STATE_4 (1 << 4)
+#define CAM_STATE_5 (1 << 5)
+#define CAM_STATE_6 (1 << 6)
+#define CAM_STATE_7 (1 << 7)
+#define CAM_STATE_UNDERWATER (1 << 8)
+#define CAM_STATE_9 (1 << 9)
+#define CAM_STATE_10 (1 << 10) // Surpresses the camera from changing settings based on the bg surface
+#define CAM_STATE_11 (1 << 11)
+#define CAM_STATE_12 (1 << 12)
+#define CAM_STATE_13 (1 << 13)
+#define CAM_STATE_14 (1 << 14)
+#define CAM_STATE_15 ((s16)(1 << 15))
```

Comment:
Sorry if you've already answered it before, but curious what is the cast for?

Reply:
Basically for matching purposes. Every use-instance of this flag in both OoT and MM needs the cast

Reply:
In theory the outer set of () shouldn't be needed here, I think? A cast only works on the () immediately following.